# stop information

In [1]:
import pandas as pd
import numpy as py

In [2]:
stops=pd.read_csv('../data_analysis/initial_data/gtfs_data/Stops.csv',index_col=0)

In [3]:
stops.head()

StopID                     StopName        Lat       Lon  StopNumber
0  8220DB000002  Parnell Square West, stop 2  53.352244 -6.263723           2
1  8220DB000003  Parnell Square West, stop 3  53.352309 -6.263811           3
2  8220DB000004  Parnell Square West, stop 4  53.352575 -6.264175           4
3  8220DB000006  Parnell Square West, stop 6  53.352749 -6.264454           6
4  8220DB000007  Parnell Square West, stop 7  53.352841 -6.264570           7

# split stopName

In [6]:
def split_stopName(stopName):
    result=stopName.split(',')
    return result[0]

stops['StopName']=stops['StopName'].map(split_stopName)
stops.head()

StopID             StopName        Lat       Lon  StopNumber
0  8220DB000002  Parnell Square West  53.352244 -6.263723           2
1  8220DB000003  Parnell Square West  53.352309 -6.263811           3
2  8220DB000004  Parnell Square West  53.352575 -6.264175           4
3  8220DB000006  Parnell Square West  53.352749 -6.264454           6
4  8220DB000007  Parnell Square West  53.352841 -6.264570           7

In [5]:
stops.head()

StopID             StopName        Lat       Lon  StopNumber
0  8220DB000002  Parnell Square West  53.352244 -6.263723           2
1  8220DB000003  Parnell Square West  53.352309 -6.263811           3
2  8220DB000004  Parnell Square West  53.352575 -6.264175           4
3  8220DB000006  Parnell Square West  53.352749 -6.264454           6
4  8220DB000007  Parnell Square West  53.352841 -6.264570           7

In [8]:
stops_by_route=pd.read_json("../data/stops_by_route.json")

In [41]:
def routes_info(stopID):
    direction=['outbound','inbound']
    lineID=stops_by_route['routes'].keys()
    result=[]
    for line in lineID:
        flag=False
        for direc in direction:
            stops_number=len(stops_by_route['routes'][line][direc])
            for stop in range(stops_number):
                if stops_by_route['routes'][line][direc][stop]["stopNumber"]==str(stopID):
                    result.append(line)
                    flag=True
                    break
            if flag:
                break
    return result       

stops['routes']=stops['StopNumber'].map(routes_info)
stops.head(5)  

StopName        Lat       Lon  StopNumber  \
0  Parnell Square West  53.352244 -6.263723           2   
1  Parnell Square West  53.352309 -6.263811           3   
2  Parnell Square West  53.352575 -6.264175           4   
3  Parnell Square West  53.352749 -6.264454           6   
4  Parnell Square West  53.352841 -6.264570           7   

                     routes  
0  [38, 38a, 38b, 46a, 46e]  
1                [120, 122]  
2        [7, 7a, 7b, 7d, 9]  
3                  [155, 4]  
4   [13, 140, 40, 40b, 40d]

In [117]:
stops.to_csv('../data_analysis/clean_data/stop_info.csv')

# load data from offical website 

In [116]:
stops_2020=pd.read_json("../data/stops.json")

In [45]:
stopPointID_20=stops_2020.stops.values
len(stopPointID_20)

4145

In [92]:
stops.count()

StopName      4220
Lat           4220
Lon           4220
StopNumber    4220
routes        4220
dtype: int64

total stops in gfts are 4220. among them, 112 are empty (no routes go through). the available stops are 4108. offical stops are 4145. there are 37 stops diffenece